code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -0,0 +1,159 @@
+// ***************************************
+// *********** Nightfall
+// ***************************************
+
+/datum/action/xeno_action/activable/nightfall
+	name = "Nightfall"
+	action_icon_state = "nightfall"
+	ability_name = "Nightfall"
+	mechanics_text = "Shut down all electrical lights for five seconds."
+	cooldown_timer = 1 MINUTES
+	plasma_cost = 100
+	/// How far nightfall will have an effect
+	var/range = 7
+	/// How long till the lights go on again
+	var/duration = 5 SECONDS
+
+/datum/action/xeno_action/activable/nightfall/on_cooldown_finish()
+	to_chat(owner, "<span class='notice'>We gather enough mental strength to shut down lights again.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/nightfall/use_ability()
+	playsound(owner, 'sound/magic/nightfall.ogg', 50, 1)
+	for(var/atom/light AS in GLOB.nightfall_toggleable_lights)
+		if((owner.loc.z != light.loc.z) || (get_dist(owner, light) >= range))
+			continue
+		light.turn_light(null, FALSE, duration, TRUE, TRUE)
+	succeed_activate()
+	add_cooldown()
+
+
+// ***************************************
+// *********** Gravity Crush
+// ***************************************
+#define ADD_FILTER (1<<0)
+#define REMOVE_FILTER (1<<1)
+#define GRAV_CRUSH (1<<2)
```

Comment:
What is this why does it exist?

Reply:
the arguments to tell the act_on_crush proc what to do